docs/Simulation_Configuration_Guide.md

```diff
@@ -37,304 +39,464 @@ Several example configuration files are provided to help you get started.
-- **`threat_swarm_configs`**: An array of threat swarms with their configurations.
+- **`end_time`**: Specifies when the simulation should end.
+- **`time_scale`**: Adjusts the speed of the simulation.
+- **`interceptor_swarm_configs`**: A list of interceptor swarms with their configurations.
+- **`threat_swarm_configs`**: A list of threat swarms with their configurations.
 
 Each swarm configuration includes:
 
 - **`num_agents`** (for threats): Number of agents in the swarm.
-- **`dynamic_agent_config`**: Configuration for agents, including:
-  - **`agent_model`**: Reference to a static agent configuration file.
-  - **`attack_behavior`** (for threats): Reference to an attack behavior configuration file.
+- **`agent_config`**: Agent configuration, including:
+  - **`config_file`**: Reference to a static agent configuration file.
+  - **`attack_behavior_config_file`** (for threats): Reference to an attack behavior configuration file.
   - **`initial_state`** (for threats): Starting position, rotation, and velocity.
   - **`standard_deviation`** (for threats): Variability in initial states.
   - **`dynamic_config`**: Time-dependent settings (sensor configurations, flight configurations).
   - **`submunitions_config`** (for interceptors): Configuration for any submunitions deployed.
     - **`num_submunitions`**: Number of submunitions that each interceptor in the swarm will release.
+    - **`agent_config`**: Submunition agent configuration.
+
+An advantage of using Protobuf messages to define the simulation configuration is that recursive message definitions are allowed.
+As a result, it is possible to represent arbitrarily deep hierarchies of submunitions in the simulation configuration.
 
 Note that the number of agents in the swarm, the initial states, and the launch times are not specified for interceptor swarms.
 The IADS will automatically determine how many interceptors to launch and when to launch them against the incoming threats by [clustering](./Simulator_Overview.md#clustering) them.
 The carrier interceptors will then independently determine [when to release the submunitions](./Simulator_Overview.md#submunitions-release).
 
 ### Examples
 
-The simulation configurations are defined in JSON files that specify the initial setup for missiles and targets.
+The simulation configurations are defined in Protobuf text format files that specify the initial setup for missiles and targets.
 
-#### Example 1: `7_quadcopters.json`
+#### Example 1: `7_quadcopters.pbtxt`
 
 A simple simulation with one interceptor type (Hydra 70) launched against seven quadcopter threats.
 
-```json
-{
-  "endTime": 300,
-  "timeScale": 1,
-  "interceptor_swarm_configs": [
-    {
-      "dynamic_agent_config": {
-        "agent_model": "hydra70.json",
-        "dynamic_config": {
-          "sensor_config": {
-            "type": "IDEAL",
-            "frequency": 100
-          },
-          "flight_config": {
-            "augmentedPnEnabled": false
+```
```

Comment:
No syntax highlighting makes the code blocks a little hard to read.
I don't see that Shiki has `textproto` highlighting. I'll investigate further.

Reply:
Resolved by 8162485d37472e9a480bb8644cf2e91d6db5fe9e and acf31f0b37d31ea7ce89344fbdf03a10b9337a21